official/vision/beta/projects/yolo/ops/preprocessing_ops.py

```diff
@@ -0,0 +1,434 @@
+    height = height * aspect_change_height
+    boxes = tf.concat([x, y, width, height], axis=-1)
+  return image, boxes
+
+
+def random_translate(image, box, t, seed=None):
+  """Randomly translate the image and boxes.
+  Args:
+      image: a `Tensor` representing the image.
+      box: a `Tensor` represeting the boxes.
+      t: an `int` representing the translation factor
+      seed: an optional seed for tf.random operations
+  Returns:
+      image: a `Tensor` representing the augmented image.
+      box: a `Tensor` representing the augmented boxes.
+  """
+  t_x = tf.random.uniform(
+      minval=-t, maxval=t, shape=(), dtype=tf.float32, seed=seed)
+  t_y = tf.random.uniform(
+      minval=-t, maxval=t, shape=(), dtype=tf.float32, seed=seed)
+  box = translate_boxes(box, t_x, t_y)
+  image = translate_image(image, t_x, t_y)
+  return image, box
+
+
+def translate_boxes(box, translate_x, translate_y):
+  """Randomly translate the boxes.
+  Args:
+    boxes: a `Tensor` represeitng the boxes.
+    translate_x: a `Tensor` represting the translation on the x-axis.
+    translate_y: a `Tensor` represting the translation on the y-axis.
+  Returns:
+    box: a `Tensor` representing the augmented boxes.
+  """
+  with tf.name_scope('translate_boxs'):
+    x = box[..., 0] + translate_x
+    y = box[..., 1] + translate_y
+    box = tf.stack([x, y, box[..., 2], box[..., 3]], axis=-1)
+    box.set_shape([None, 4])
+  return box
+
+
+def translate_image(image, translate_x, translate_y):
+  """Randomly translate the image.
+        Args:
+            image: a `Tensor` representing the image.
+            translate_x: a `Tensor` represting the translation on the x-axis.
+            translate_y: a `Tensor` represting the translation on the y-axis.
+        Returns:
+            box: a `Tensor` representing the augmented boxes.
+        """
+  with tf.name_scope('translate_image'):
+    if (translate_x != 0 and translate_y != 0):
+      image_jitter = tf.convert_to_tensor([translate_x, translate_y])
+      image_jitter.set_shape([2])
+      image = tfa.image.translate(
+          image, image_jitter * tf.cast(tf.shape(image)[1], tf.float32))
+  return image
+
+
+def pad_max_instances(value, instances, pad_value=0, pad_axis=0):
+  shape = tf.shape(value)
+  dim1 = shape[pad_axis]
+  take = tf.math.reduce_min([instances, dim1])
+  value, _ = tf.split(
+      value, [take, -1], axis=pad_axis)  # value[:instances, ...]
+  pad = tf.convert_to_tensor([tf.math.reduce_max([instances - dim1, 0])])
+  nshape = tf.concat([shape[:pad_axis], pad, shape[(pad_axis + 1):]], axis=0)
+  pad_tensor = tf.fill(nshape, tf.cast(pad_value, dtype=value.dtype))
+  value = tf.concat([value, pad_tensor], axis=pad_axis)
+  return value
+
+
+def fit_preserve_aspect_ratio(image,
+                              boxes,
+                              width=None,
+                              height=None,
+                              target_dim=None):
```

Comment:
add docstring

Reply:
Fixed in https://github.com/tensorflow/models/pull/9493/commits/bb11fe05a6a24b084766659e149431d0f4fc1da4